components/src/dynamo/common/utils/guided_json.py

```
@@ -0,0 +1,289 @@
+                stack.pop()
+                continue
+
+            neighbor_state = state.get(neighbor, 0)
+            if neighbor_state == 0:
+                state[neighbor] = 1
+                active_indexes[neighbor] = len(active_nodes)
+                active_nodes.append(neighbor)
+                stack.append((neighbor, iter(edges.get(neighbor, ()))))
+            elif neighbor_state == 1:
+                cycle_start = active_indexes[neighbor]
+                cycle = active_nodes[cycle_start:] + [neighbor]
```

Ruff (0.16.1)

[warning] Consider `[*active_nodes[cycle_start:], neighbor]` instead of concatenation

Replace with `[*active_nodes[cycle_start:], neighbor]`

(RUF005)